### PR TITLE
Fixes for RISC-V

### PR DESCRIPTION
**Pull request by JohnAZoidberg**

### CODE REVIEW

File: BaseTools/Conf/tools_def.template [hunk 1]
```diff
@@ -1947,7 +1947,7 @@ DEFINE GCC_IA32_RC_FLAGS           = -I binary -O elf32-i386          -B i386
 DEFINE GCC_X64_RC_FLAGS            = -I binary -O elf64-x86-64        -B i386    --rename-section .data=.hii
 DEFINE GCC_ARM_RC_FLAGS            = -I binary -O elf32-littlearm     -B arm     --rename-section .data=.hii
 DEFINE GCC_AARCH64_RC_FLAGS        = -I binary -O elf64-littleaarch64 -B aarch64 --rename-section .data=.hii
-DEFINE GCC_RISCV64_RC_FLAGS        = -I binary -O elf64-littleriscv -  B riscv64 --rename-section .data=.hii
+DEFINE GCC_RISCV64_RC_FLAGS        = -I binary -O elf64-littleriscv   -B riscv   --rename-section .data=.hii
```
Comment (changab) [inline BaseTools/Conf/tools_def.template:1950]: What is the build or functionality error with riscv64

Comment (JohnAZoidberg) [reply BaseTools/Conf/tools_def.template:1950]:
```
$ riscv64-unknown-elf-objcopy -B riscv64
riscv64-unknown-elf-objcopy: architecture riscv64 unknown
```

File: RiscVPkg/Library/RiscVDxeIplHandoffOpenSbiLib/RiscVDxeIplHandoffOpenSbiLib.c [hunk 2]
```diff
@@ -18,22 +18,22 @@
...
 
 /**
-   RISC-V platform DXE IPL to DXE OpenSBI mdoe switch handler.
+   RISC-V platform DXE IPL to DXE OpenSBI mode switch handler.
    This function is executed in RISC-V Supervisor mode.
+   It is called by sbi_init()
 
    This function performs a CPU architecture specific operations to execute
    the entry point of DxeCore with the parameters of HobList.
+   // Question: Doesn't seem to install it, does it?
```
Comment (JohnAZoidberg) [inline RiscVPkg/Library/RiscVDxeIplHandoffOpenSbiLib/RiscVDxeIplHandoffOpenSbiLib.c]: cc @changab 
Is that missing?

File: BaseTools/Conf/tools_def.template (hunk 1 above)
Comment (changab) [reply BaseTools/Conf/tools_def.template:1950]: I have to figure out why it doesn't happen on my build.

File: RiscVPkg/Library/RiscVDxeIplHandoffOpenSbiLib/RiscVDxeIplHandoffOpenSbiLib.c (hunk 2 above)
Comment (changab) [reply RiscVPkg/Library/RiscVDxeIplHandoffOpenSbiLib/RiscVDxeIplHandoffOpenSbiLib.c]: Ah yes.
we missed installing END_OF_PEI.
PeiServicesInstallPpi (&gEndOfPeiSignalPpi);

File: BaseTools/Conf/tools_def.template (hunk 1 above)
Comment (JohnAZoidberg) [reply BaseTools/Conf/tools_def.template:1950]: Ah, I think this error was not triggered for the PPI build.
I think I can trigger it when I build `MdeModulePkg/MdeModulePkg.dsc`.

Comment (changab) [reply BaseTools/Conf/tools_def.template:1950]: ok. The error happens when build MdeModulePkg/MdeModulePkg.dsc alone, right?

Comment (JohnAZoidberg) [reply BaseTools/Conf/tools_def.template:1950]: Yep. I wanted to build the HelloWorld shell application.

File: RiscVPkg/Library/RiscVDxeIplHandoffOpenSbiLib/RiscVDxeIplHandoffOpenSbiLib.c (hunk 2 above)
Comment (JohnAZoidberg) [reply RiscVPkg/Library/RiscVDxeIplHandoffOpenSbiLib/RiscVDxeIplHandoffOpenSbiLib.c]: Hm, when I add it, it gets stuck there. I'll investigate that. WIP here: #4

**Comment by JohnAZoidberg** [comment]
@changab Is this good?

I put `iscVDxeIplHandoffOpenSbiLib: Install gEndOfPeiSignalPpi` here and rebased my other branches on top of this.

**Comment by changab** [comment]
yes. it looks good.

File: BaseTools/Conf/tools_def.template (hunk 1 above)
Comment (changab) [reply BaseTools/Conf/tools_def.template:1950]: I can just add RISCV64 to 
  SUPPORTED_ARCHITECTURES        = IA32|X64|EBC|ARM|AARCH64|`RISCV64` in MdeModulePkg.dsc and MdePkg.dsc then the error you met is gone.
However, I still get build error on MdeModulePkg due to gEfiMdeModulePkgTokenSpaceGuid.PcdDxeIplSwitchToLongMode is not declared in RISC-V arch. This is edk2 problem but our problem. PcdDxeIplSwitchToLongMode is only for x86 system. I will just leave this error there, not dealing with it just now.

**Comment by changab** [comment]
Please do followings,
1. Revert the changes on BaseTools/Conf/tools_def.template.
2. Add RISCV64 to MdeModulePkg.dsc and MdePkg.dsc.
3 Commit again

**Comment by JohnAZoidberg** [comment]
> Please do the following
> 1. Revert the changes on BaseTools/Conf/tools_def.template.
> 2. Add RISCV64 to MdeModulePkg.dsc and MdePkg.dsc.

I kept 1 because it's necessary for #5.
And I added 2 in #5.

File: BaseTools/Conf/tools_def.template (hunk 1 above)
Comment (JohnAZoidberg) [reply BaseTools/Conf/tools_def.template:1950]: I updated my compiler toolchain and it's still the case that we need `riscv` not `riscv64` here.

Comment (JohnAZoidberg) [reply BaseTools/Conf/tools_def.template:1950]: @changab You didn't squash this in any commit?

Comment (changab) [reply BaseTools/Conf/tools_def.template:1950]: No. I will have to check this. just have no time. Will do it soon after I clean up some tasks.
we don't have 'riscv' architecture BTW, we only have either RISCV64 or RISCV32. RISCV32 has been killed from edk2 RISC-V port coz we don't even verify that on QEMU or board.
Give me some time to back on this.

**Pull request closed without merging**